Курсовой проект. Часть 3. Первый вариант.

aopanasenko opened the pull request:
Сделаны функции по решению уравнения Кеплера, а так же нахождению координат спутника в заданный момент времени. В файле main.cpp вычисляются положения спутника в течение суток. При попытке собрать проект в утилите CMake выдается ошибка "Could NOT find CHECK" после чего тест благополучно пропускается.

aopanasenko committed:
Курсовой проект. Часть 3. Первый вариант.

Сделаны функции по решению уравнения Кеплера, а так же нахождению координат спутника в заданный момент времени. В файле main.cpp вычисляются положения спутника в течение суток. При попытке собрать проект в утилите CMake выдается ошибка "Could NOT find CHECK" после чего тест благополучно пропускается.

Korogodin commented:
Вероятно надо.....поставить check? См. README.md в корне библиотеки

aopanasenko committed:
Delete CMakeLists.txt

aopanasenko committed:
Delete AddCompilerFlag.cmake

aopanasenko committed:
Delete CheckCCompilerFlag.cmake

aopanasenko committed:
Delete CheckCXXCompilerFlag.cmake

aopanasenko committed:
Delete FindCheck.cmake

aopanasenko committed:
Delete Toolchain-gcc-arm-linux.cmake

aopanasenko committed:
Delete CMakeLists.txt

aopanasenko committed:
Delete check_main.cpp

aopanasenko committed:
Delete check_position.cpp

aopanasenko committed:
Delete check_suites.h

aopanasenko committed:
Delete CMakeLists.txt

aopanasenko committed:
Delete gpssvpos.cpp

aopanasenko committed:
Delete kepler.cpp

aopanasenko committed:
Delete main.cpp

aopanasenko committed:
Delete CMakeCache.txt

aopanasenko committed:
Delete CMakeOutput.log

aopanasenko committed:
Delete cmake.check_cache

aopanasenko committed:
Delete feature_tests.bin

aopanasenko committed:
Delete feature_tests.c

aopanasenko committed:
Delete feature_tests.cxx

aopanasenko committed:
Delete CMakeCCompiler.cmake

aopanasenko committed:
Delete VCTargetsPath.vcxproj

aopanasenko committed:
Delete VCTargetsPath.txt

aopanasenko committed:
Delete CMakeSystem.cmake

aopanasenko committed:
Delete CMakeRCCompiler.cmake

aopanasenko committed:
Delete CMakeDetermineCompilerABI_CXX.bin

aopanasenko committed:
Delete CMakeDetermineCompilerABI_C.bin

aopanasenko committed:
Delete CMakeCXXCompiler.cmake

aopanasenko committed:
Delete VCTargetsPath.lastbuildstate

aopanasenko committed:
Delete CMakeCXXCompilerId.cpp

aopanasenko committed:
Delete CompilerIdCXX.exe

aopanasenko committed:
Delete CompilerIdCXX.vcxproj

aopanasenko committed:
Delete CMakeCXXCompilerId.obj

aopanasenko committed:
Delete CL.command.1.tlog

aopanasenko committed:
Delete CL.read.1.tlog

aopanasenko committed:
Delete CL.write.1.tlog

aopanasenko committed:
Delete CompilerIdCXX.lastbuildstate

aopanasenko committed:
Delete link.command.1.tlog

aopanasenko committed:
Delete link.read.1.tlog

aopanasenko committed:
Delete link.write.1.tlog

aopanasenko committed:
Delete CMakeCCompilerId.c

aopanasenko committed:
Delete CompilerIdC.exe

aopanasenko committed:
Delete CompilerIdC.vcxproj

aopanasenko committed:
Delete CMakeCCompilerId.obj

aopanasenko committed:
Delete gpssvpos.h

aopanasenko committed:
Delete kepler.h

aopanasenko committed:
Delete link.write.1.tlog

aopanasenko committed:
Delete link.read.1.tlog

aopanasenko committed:
Delete link.command.1.tlog

aopanasenko committed:
Delete CompilerIdC.lastbuildstate

aopanasenko committed:
Delete CL.write.1.tlog

aopanasenko committed:
Delete CL.read.1.tlog

aopanasenko committed:
Delete CL.command.1.tlog

aopanasenko committed:
Delete rules.ninja

aopanasenko committed:
Delete cmake_install.cmake

aopanasenko committed:
Delete build.ninja

aopanasenko committed:
Delete VSInheritEnvironments.txt

aopanasenko committed:
Delete CMakeCache.txt

aopanasenko committed:
Delete cmake_install.cmake

aopanasenko committed:
Delete cmake_install.cmake

aopanasenko committed:
Delete LastTest.log

aopanasenko committed:
Delete CMakeOutput.log

aopanasenko committed:
Delete TargetDirectories.txt

aopanasenko committed:
Delete cmake.check_cache

aopanasenko committed:
Delete feature_tests.bin

aopanasenko committed:
Delete feature_tests.c

aopanasenko committed:
Delete feature_tests.cxx

aopanasenko committed:
Delete main.obj

aopanasenko committed:
Delete main.c

aopanasenko committed:
Delete foo.h

aopanasenko committed:
Delete CMakeCCompiler.cmake

aopanasenko committed:
Delete CMakeCXXCompiler.cmake

aopanasenko committed:
Delete CMakeDetermineCompilerABI_C.bin

aopanasenko committed:
Delete CMakeDetermineCompilerABI_CXX.bin

aopanasenko committed:
Delete CMakeRCCompiler.cmake

aopanasenko committed:
Delete CMakeSystem.cmake

aopanasenko committed:
Delete CMakeCXXCompilerId.cpp

aopanasenko committed:
Delete CMakeCXXCompilerId.exe

aopanasenko committed:
Delete CMakeCXXCompilerId.obj

aopanasenko committed:
Delete CMakeCCompilerId.c

aopanasenko committed:
Delete CMakeCCompilerId.exe

aopanasenko committed:
Delete CMakeCCompilerId.obj

aopanasenko committed:
Курсовой проект. Часть 3. Исправление 1.

Нашел способ как запустить тесты под виндой. Ошибок при сборке нет. Исключения не выбрасываются. Тесты не проходятся. Пробовал двумя методами: сравнивать каждую координату с координатой из матлаба и сравнение сумм координат.

aopanasenko commented:
Нашел таки способ как в MVS сделать тесты: https://www.bestprog.net/ru/2018/08/16/example-of-unit-test-creating-in-microsoft-visual-studio-c_ru/

Korogodin commented:
Идеально значения из матлаба не будут совпадать со значениями в Си: может использоваться разная разрядность для представления чисел, по-разному реализованы тригонометрические функции, результат может огрубляться при записи в файл и т.п. Поэтому надо добиться, чтобы результаты совпадали с приемлемой с практической точки зрения точностью. Скажем, с точностью в миллиметр. 

Смотри какая координата и в какое время не сходится, далее сравнивай пошагово расчет в матлабе и в Си.

aopanasenko committed:
Курсовой проект 3. Исправление №2.

Тесты проходятся, утечек памяти нет, исключения не выбрасываются. Для контроля утечек памяти использовалась библиотека Microsoft - CRT.

Korogodin commented:
Проверять координаты с точностью в 100 км -- несколько самонадеянно. Но да ладно, собирай итоговый отчёт, прикрепляй, открывай новый PR для него

Korogodin commented:
Итоговая оценка отл, проставлена в ведомость и зачетку, бумажная версия отчета сдана
